planner/core/logical_plan_builder.go

```diff
@@ -687,19 +686,23 @@ func (b *PlanBuilder) buildSelection(ctx context.Context, p LogicalPlan, where a
 
 // buildProjectionFieldNameFromColumns builds the field name, table name and database name when field expression is a column reference.
 func (b *PlanBuilder) buildProjectionFieldNameFromColumns(origField *ast.SelectField, colNameField *ast.ColumnNameExpr, name *types.FieldName) (colName, origColName, tblName, origTblName, dbName model.CIStr) {
-	origColName, tblName, dbName = colNameField.Name.Name, colNameField.Name.Table, colNameField.Name.Schema
-	if origField.AsName.L != "" {
-		colName = origField.AsName
+	origTblName, dbName = name.OrigTblName, name.DBName
+	if name.OrigColName.L == "" {
```

Comment:
if we set original column here https://github.com/pingcap/tidb/pull/https://github.com/pingcap/tidb/blob/3ebc09631dbc700100f1c30617878f7c6c46c888/planner/core/logical_plan_builder.go#diff-5f46fc7e65bc1be2746caf2464f58d13L616
do we still need this if branch here?

Reply:
@winoros comment addressed, PTAL

Reply:
Yes, `ColumnInfos2ColumnsAndNames` function is called when build insert plan, while
`buildProjectionFieldNameFromColumns` function is called when build select plan.
if we do not set `OrigTblName` in `ColumnInfos2ColumnsAndNames`, this SQL will throw an error:
```
create table t1(a int primary key, b int default 20, c int default 30);
insert into t1 set a = 1, b = default(c); # <--[table:1364]Field 'c' doesn't have a default value
```